src/publishMain.ml

```diff
@@ -610,7 +695,16 @@ let main_term root =
     let meta_opams =
       get_opams force dirs opams urls projects tag names version
     in
-    if not (OpamConsole.confirm
+    let default =
+      if not skip_deps_check then
+        let missing_direct_deps, pinned_pkgs =
+          check_depends_stanza meta_opams opam_root
+        in
+        not (missing_direct_deps <> [] || pinned_pkgs <> [])
```

Comment:
Why not returning directly the boolean? In that case, you can have only `skip_deps_check || check_depends_stanza`